changedetectionio/__init__.py

```diff
@@ -744,6 +744,27 @@ def preview_page(uuid):
                                  uuid=uuid)
         return output
 
+    @app.route("/api/<string:uuid>/current_snapshot", methods=['GET'])
+    @login_required
+    def api_content(uuid):
+
+        # More for testing, possible to return the first/only
+        if uuid == 'first':
+            uuid = list(datastore.data['watching'].keys()).pop()
+
+        try:
+            watch = datastore.data['watching'][uuid]
+        except KeyError:
+            flash("No history found for the specified link, bad link?", "error")
```

Comment:
For API's, it should not redirect (302 or 200), it should give some more obvious error

https://www.baeldung.com/rest-api-error-handling-best-practices

I think `400` is best here

> 400-level (Client error) – client sent an invalid request

Reply:
Done